date-time/src/commonMain/kotlin/com/splendo/kaluga/datetime/timer/Timer.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ Copyright 2021 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+package com.splendo.kaluga.datetime.timer
+
+import com.splendo.kaluga.base.utils.firstInstance
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.flowOf
+import kotlin.time.Duration
+
+/** A timer ticking a certain [duration]. */
+interface Timer {
+    /** Timer duration. */
+    val duration: Duration
+    /** Current state of the timer. */
+    val state: StateFlow<State>
+
+    /** Starts the timer. */
+    suspend fun start()
```

Comment:
Is it really needs to be suspend? (same for pause)

Reply:
on the second thought, there is not much benefit. updating to not suspending functions